spring-integration-core/src/main/java/org/springframework/integration/support/management/graph/IntegrationGraphServer.java

```diff
@@ -162,6 +164,19 @@ private void gateways(Collection<IntegrationNode> nodes, Collection<LinkNode> li
 			nodes.add(gatewayNode);
 			producerLink(links, channelNodes, gatewayNode);
 		}
+		Map<String, GatewayProxyFactoryBean> gpfbs = this.applicationContext
+				.getBeansOfType(GatewayProxyFactoryBean.class);
+		for (Entry<String, GatewayProxyFactoryBean> entry : gpfbs.entrySet()) {
+			Map<Method, MessagingGatewaySupport> methodMap = entry.getValue().getGateways();
+			int i = 0;
+			for (Entry<Method, MessagingGatewaySupport> gwEntry : methodMap.entrySet()) {
+				MessagingGatewaySupport gateway = gwEntry.getValue();
+				MessageGatewayNode gatewayNode = this.nodeFactory.gatewayNode(
+						entry.getKey().substring(1) + "." + gwEntry.getKey().getName() + "#" + i++, gateway);
```

Comment:
M-m-m. Not sure in the `i++` for each method:
```
{
    "nodeId" : 10,
    "name" : "integrationGraphServerTests$Gate.bar#0",
    "stats" : null,
    "componentType" : "gateway",
    "output" : "four",
    "errors" : null
  }, {
    "nodeId" : 11,
    "name" : "integrationGraphServerTests$Gate.foo#1",
    "stats" : null,
    "componentType" : "gateway",
    "output" : "four",
    "errors" : null
  }
```
These methods are unique, so no need in the index for them.
How about to produce a method signature? 
That is how they are distinguished in Java:
```
Method method = gwEntry.getKey();
String signature = method.getName() +
		"(" + StringUtils.arrayToCommaDelimitedString(method.getParameterTypes()) + ")";
MessageGatewayNode gatewayNode = this.nodeFactory.gatewayNode(entry.getKey().substring(1) + "." + signature, gateway);
```
Where we have a result like:
```
 {
    "nodeId" : 10,
    "name" : "integrationGraphServerTests$Gate.bar(class java.lang.String)",
    "stats" : null,
    "componentType" : "gateway",
    "output" : "four",
    "errors" : null
  }, {
    "nodeId" : 11,
    "name" : "integrationGraphServerTests$Gate.foo(class java.lang.Integer)",
    "stats" : null,
    "componentType" : "gateway",
    "output" : "four",
    "errors" : null
  }, {
    "nodeId" : 12,
    "name" : "integrationGraphServerTests$Gate.foo(class java.lang.String)",
    "stats" : null,
    "componentType" : "gateway",
    "output" : "four",
    "errors" : null
  }
```
Yes, I understand that it is so long, but we have there tooltip to see more.
CC @aclement